[Bug] GitVersionInformation.g.cs(12,42): error CS0234 'ExcludeFromCodeCoverageAttribute' does not exist

**Describe the bug**

Latest version 5.3.6 fails in `.Net Standard 2.0` builds.

## Expected Behavior

Expected build to not fail. Version 5.3.5 builds all frameworks correctly.

## Actual Behavior

During build of a dll in `.Net Standard 2.0` I am getting following error:

```bash
  Unity.Abstractions -> D:\a\abstractions\abstractions\src\bin\Release\netstandard2.0\Unity.Abstractions.dll
obj\Release\netstandard1.6\GitVersionInformation.g.cs(12,42): error CS0234: The type or namespace name 'ExcludeFromCodeCoverageAttribute' does not exist in the namespace 'System.Diagnostics.CodeAnalysis' (are you missing an assembly reference?) [D:\a\abstractions\abstractions\src\Unity.Abstractions.csproj]
```

The generated file fails on line 12:

```C#
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     GitVersion
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

[global::System.Runtime.CompilerServices.CompilerGenerated]
[global::System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]   <-- This line fails in NETSTANDARD2_0
static class GitVersionInformation
{
    public static string Major = "6";
    public static string Minor = "0";
    public static string Patch = "0";
. . .
}
```

## Possible Fix

Reverted to previous version for now

## Steps to Reproduce

For more information please follow [this link](https://github.com/unitycontainer/abstractions/runs/778004700?check_suite_focus=true) and examine `Build` step.

## Context

I am using GitVerion tool in CI builds. The library is added to projects during CI steps

## Your Environment

<TargetFrameworks>netstandard2.1;netstandard2.0;netstandard1.6;netcoreapp3.0;netcoreapp2.0;netcoreapp1.0;net48;net47;net46;net45;net40</TargetFrameworks>

- Version Used: 5.3.6
- Link to your project: https://github.com/unitycontainer/abstractions
- Link to your CI build (if appropriate): https://github.com/unitycontainer/abstractions/blob/release/6.0.0/.github/workflows/cd-flow.yml

@ENikS 

> During build of a dll in .Net Standard 2.0 I am getting following error:
>  Unity.Abstractions -> D:\a\abstractions\abstractions\src\bin\Release\netstandard2.0\Unity.Abstractions.dll
obj\Release\netstandard1.6\GitVersionInformation.g.cs(12,42): error CS0234: The type or namespace name 'ExcludeFromCodeCoverageAttribute' does not exist in the namespace 'System.Diagnostics.CodeAnalysis' (are you missing an assembly reference?) [D:\a\abstractions\abstractions\src\Unity.Abstractions.csproj]

As can be seen in the logs you provided us, the issue seems to be rather related to .NET Standard **1.6**, not **2.0**: if you examine the path in the log, you have _netstandard1.6_ after _obj\Release_ (there may be something weird in `Unity.Abstractions.csproj` by the way).

I double-checked in [msdn](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.excludefromcodecoverageattribute?view=netcore-3.1) what frameworks support the attribute and indeed it is unknown by .NET Standard < 2.0...

Still, we'll need some `#if` on our side, but I honestly expect it to work when 'really' targeting .NET Standard 2.0; I'll go clone your repository and look for this!

--- a/src/GitVersionCore.Tests/VersionConverters/Approved/cs/GitVersionInfoGeneratorTests.ShouldCreateFile.approved.txt
+++ b/src/GitVersionCore.Tests/VersionConverters/Approved/cs/GitVersionInfoGeneratorTests.ShouldCreateFile.approved.txt
@@ -8,6 +8,22 @@
 // </auto-generated>
 //------------------------------------------------------------------------------
 
+#if NET20 || NET35 || NETCOREAPP1_0 || NETCOREAPP1_1 || NETSTANDARD1_0 || NETSTANDARD1_1 || NETSTANDARD1_2 || NETSTANDARD1_3 || NETSTANDARD1_4 || NETSTANDARD1_5 || NETSTANDARD1_6
+namespace System.Diagnostics.CodeAnalysis
+{
+    [global::System.AttributeUsage(
+        global::System.AttributeTargets.Assembly |
+        global::System.AttributeTargets.Class |
+        global::System.AttributeTargets.Struct |
+        global::System.AttributeTargets.Constructor |
+        global::System.AttributeTargets.Method |
+        global::System.AttributeTargets.Property |
+        global::System.AttributeTargets.Event,
+        Inherited = false, AllowMultiple = false)]
+    internal sealed class ExcludeFromCodeCoverageAttribute : global::System.Attribute { }
+}
+#endif
+
 [global::System.Runtime.CompilerServices.CompilerGenerated]
 [global::System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
 static class GitVersionInformation

--- a/src/GitVersionCore.Tests/VersionConverters/Approved/fs/GitVersionInfoGeneratorTests.ShouldCreateFile.approved.txt
+++ b/src/GitVersionCore.Tests/VersionConverters/Approved/fs/GitVersionInfoGeneratorTests.ShouldCreateFile.approved.txt
@@ -8,6 +8,22 @@
 // </auto-generated>
 //------------------------------------------------------------------------------
 
+#if NET20 || NET35 || NETCOREAPP1_0 || NETCOREAPP1_1 || NETSTANDARD1_0 || NETSTANDARD1_1 || NETSTANDARD1_2 || NETSTANDARD1_3 || NETSTANDARD1_4 || NETSTANDARD1_5 || NETSTANDARD1_6
+namespace System.Diagnostics.CodeAnalysis
+
+[<Sealed>]
+[<global.System.AttributeUsage(
+    global.System.AttributeTargets.Assembly ||| 
+    global.System.AttributeTargets.Class ||| 
+    global.System.AttributeTargets.Struct ||| 
+    global.System.AttributeTargets.Constructor ||| 
+    global.System.AttributeTargets.Method ||| 
+    global.System.AttributeTargets.Property ||| 
+    global.System.AttributeTargets.Event, 
+    Inherited = false, AllowMultiple = false)>]
+type ExcludeFromCodeCoverageAttribute() = inherit global.System.Attribute()
+#endif
+
 namespace global
 
 [<AbstractClass; Sealed>]

--- a/src/GitVersionCore.Tests/VersionConverters/Approved/vb/GitVersionInfoGeneratorTests.ShouldCreateFile.approved.txt
+++ b/src/GitVersionCore.Tests/VersionConverters/Approved/vb/GitVersionInfoGeneratorTests.ShouldCreateFile.approved.txt
@@ -8,6 +8,22 @@
 ' </auto-generated>
 '------------------------------------------------------------------------------
 
+#If NET20 OrElse NET35 OrElse NETCOREAPP1_0 OrElse NETCOREAPP1_1 OrElse NETSTANDARD1_0 OrElse NETSTANDARD1_1 OrElse NETSTANDARD1_2 OrElse NETSTANDARD1_3 OrElse NETSTANDARD1_4 OrElse NETSTANDARD1_5 OrElse NETSTANDARD1_6 Then
+Namespace Global.System.Diagnostics.CodeAnalysis
+    <Global.System.AttributeUsage(
+        Global.System.AttributeTargets.Assembly Or
+        Global.System.AttributeTargets.Class Or
+        Global.System.AttributeTargets.Struct Or
+        Global.System.AttributeTargets.Constructor Or
+        Global.System.AttributeTargets.Method Or
+        Global.System.AttributeTargets.Property Or
+        Global.System.AttributeTargets.Event, Inherited:=False, AllowMultiple:=False)>
+    Friend NotInheritable Class ExcludeFromCodeCoverageAttribute
+        Inherits Global.System.Attribute
+    End Class
+End Namespace
+#End If
+
 Namespace Global
 
     <Global.System.Runtime.CompilerServices.CompilerGenerated()>

--- a/src/GitVersionCore/VersionConverters/GitVersionInfo/Templates/GitVersionInformation.cs
+++ b/src/GitVersionCore/VersionConverters/GitVersionInfo/Templates/GitVersionInformation.cs
@@ -8,6 +8,22 @@
 // </auto-generated>
 //------------------------------------------------------------------------------
 
+#if NET20 || NET35 || NETCOREAPP1_0 || NETCOREAPP1_1 || NETSTANDARD1_0 || NETSTANDARD1_1 || NETSTANDARD1_2 || NETSTANDARD1_3 || NETSTANDARD1_4 || NETSTANDARD1_5 || NETSTANDARD1_6
+namespace System.Diagnostics.CodeAnalysis
+{{
+    [global::System.AttributeUsage(
+        global::System.AttributeTargets.Assembly |
+        global::System.AttributeTargets.Class |
+        global::System.AttributeTargets.Struct |
+        global::System.AttributeTargets.Constructor |
+        global::System.AttributeTargets.Method |
+        global::System.AttributeTargets.Property |
+        global::System.AttributeTargets.Event,
+        Inherited = false, AllowMultiple = false)]
+    internal sealed class ExcludeFromCodeCoverageAttribute : global::System.Attribute {{ }}
+}}
+#endif
+
 [global::System.Runtime.CompilerServices.CompilerGenerated]
 [global::System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
 static class GitVersionInformation

--- a/src/GitVersionCore/VersionConverters/GitVersionInfo/Templates/GitVersionInformation.fs
+++ b/src/GitVersionCore/VersionConverters/GitVersionInfo/Templates/GitVersionInformation.fs
@@ -8,6 +8,22 @@
 // </auto-generated>
 //------------------------------------------------------------------------------
 
+#if NET20 || NET35 || NETCOREAPP1_0 || NETCOREAPP1_1 || NETSTANDARD1_0 || NETSTANDARD1_1 || NETSTANDARD1_2 || NETSTANDARD1_3 || NETSTANDARD1_4 || NETSTANDARD1_5 || NETSTANDARD1_6
+namespace System.Diagnostics.CodeAnalysis
+
+[<Sealed>]
+[<global.System.AttributeUsage(
+    global.System.AttributeTargets.Assembly ||| 
+    global.System.AttributeTargets.Class ||| 
+    global.System.AttributeTargets.Struct ||| 
+    global.System.AttributeTargets.Constructor ||| 
+    global.System.AttributeTargets.Method ||| 
+    global.System.AttributeTargets.Property ||| 
+    global.System.AttributeTargets.Event, 
+    Inherited = false, AllowMultiple = false)>]
+type ExcludeFromCodeCoverageAttribute() = inherit global.System.Attribute()
+#endif
+
 namespace global
 
 [<AbstractClass; Sealed>]

--- a/src/GitVersionCore/VersionConverters/GitVersionInfo/Templates/GitVersionInformation.vb
+++ b/src/GitVersionCore/VersionConverters/GitVersionInfo/Templates/GitVersionInformation.vb
@@ -8,6 +8,22 @@
 ' </auto-generated>
 '------------------------------------------------------------------------------
 
+#If NET20 OrElse NET35 OrElse NETCOREAPP1_0 OrElse NETCOREAPP1_1 OrElse NETSTANDARD1_0 OrElse NETSTANDARD1_1 OrElse NETSTANDARD1_2 OrElse NETSTANDARD1_3 OrElse NETSTANDARD1_4 OrElse NETSTANDARD1_5 OrElse NETSTANDARD1_6 Then
+Namespace Global.System.Diagnostics.CodeAnalysis
+    <Global.System.AttributeUsage(
+        Global.System.AttributeTargets.Assembly Or
+        Global.System.AttributeTargets.Class Or
+        Global.System.AttributeTargets.Struct Or
+        Global.System.AttributeTargets.Constructor Or
+        Global.System.AttributeTargets.Method Or
+        Global.System.AttributeTargets.Property Or
+        Global.System.AttributeTargets.Event, Inherited:=False, AllowMultiple:=False)>
+    Friend NotInheritable Class ExcludeFromCodeCoverageAttribute
+        Inherits Global.System.Attribute
+    End Class
+End Namespace
+#End If
+
 Namespace Global
 
     <Global.System.Runtime.CompilerServices.CompilerGenerated()>

